sonar-core/src/main/java/org/sonar/core/util/DefaultHttpDownloader.java

```diff
@@ -197,6 +201,13 @@ private void initProxy(SystemFacade system, Settings settings) {
           system.setProperty(key, settings.getString(key));
         }
       }
+      // defaults of HTTPS properties are the values of HTTP properties
+      if (!settings.hasKey(HTTPS_PROXY_HOST) && settings.hasKey(HTTP_PROXY_HOST)) {
+        system.setProperty(HTTPS_PROXY_HOST, settings.getString(HTTP_PROXY_HOST));
+      }
+      if (!settings.hasKey(HTTPS_PROXY_PORT) && settings.hasKey(HTTP_PROXY_PORT)) {
+        system.setProperty(HTTPS_PROXY_PORT, settings.getString(HTTP_PROXY_PORT));
+      }
```

Comment:
You could create a method setPropertyToDefaultIfNotSet (or something different)  in order to prevent duplication.